features/dd-sdk-android-trace/src/main/java/com/datadog/trace/api/metrics/SpanMetricRegistryImpl.java

```diff
@@ -1,48 +1,52 @@
 package com.datadog.trace.api.metrics;
 
+import com.datadog.trace.core.util.MapUtils;
+
 import java.util.Collection;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
-/** The default {@link SpanMetricRegistry} implementation. */
+/**
+ * The default {@link SpanMetricRegistry} implementation.
+ */
 public class SpanMetricRegistryImpl implements SpanMetricRegistry {
-  private static final SpanMetricRegistryImpl INSTANCE = new SpanMetricRegistryImpl();
-  private final Map<String, SpanMetricsImpl> spanMetrics;
-
-  public static SpanMetricRegistryImpl getInstance() {
-    return INSTANCE;
-  }
-
-  private SpanMetricRegistryImpl() {
-    this.spanMetrics = new ConcurrentHashMap<>();
-  }
-
-  @Override
-  public SpanMetrics get(String instrumentationName) {
-    return this.spanMetrics.computeIfAbsent(instrumentationName, SpanMetricsImpl::new);
-  }
-
-  @Override
-  public String summary() {
-    StringBuilder summary = new StringBuilder();
-    for (SpanMetricsImpl metric : spanMetrics.values()) {
-      summary.append(metric.getInstrumentationName());
-      String separator = ": ";
-      for (CoreCounter counter : metric.getCounters()) {
-        summary.append(separator).append(counter.getName()).append('=').append(counter.getValue());
-        separator = ", ";
-      }
-      summary.append('\n');
+    private static final SpanMetricRegistryImpl INSTANCE = new SpanMetricRegistryImpl();
+    private final Map<String, SpanMetricsImpl> spanMetrics;
+
+    public static SpanMetricRegistryImpl getInstance() {
+        return INSTANCE;
+    }
+
+    private SpanMetricRegistryImpl() {
+        this.spanMetrics = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public SpanMetrics get(String instrumentationName) {
+        return MapUtils.computeIfAbsent(instrumentationName, spanMetrics, SpanMetricsImpl::new);
```

Comment:
such form of constructor invocation `SpanMetricsImpl::new` is still compatible with Java 7?

Reply:
lint doesn't say anything...I will double check

Reply:
it seems so...compiler and lint is not complaining at all it is actually suggesting me to use that in case I use the function implementation itself.